node-graph/gsvg-renderer/src/renderer.rs

```diff
@@ -1036,70 +1147,188 @@ impl Render for Table<Vector> {
-						let weight = row.element.style.stroke().as_ref().map_or(0., Stroke::effective_width);
-						let quad = Quad::from_box(bounds).inflate(weight * max_scale(applied_stroke_transform));
-						let bounds = quad.bounding_box();
-						let rect = kurbo::Rect::new(bounds[0].x, bounds[0].y, bounds[1].x, bounds[1].y);
-
-						let compose = if row.element.style.stroke().is_some_and(|x| x.align == StrokeAlign::Outside) {
-							peniko::Compose::SrcOut
-						} else {
-							peniko::Compose::SrcIn
-						};
-
-						if wants_stroke_below {
-							scene.push_layer(peniko::Mix::Normal, 1., kurbo::Affine::IDENTITY, &rect);
-							vector_table.render_to_vello(scene, parent_transform, _context, &render_params.for_alignment(applied_stroke_transform));
-							scene.push_layer(peniko::BlendMode::new(peniko::Mix::Normal, compose), 1., kurbo::Affine::IDENTITY, &rect);
+					if is_branching {
+						// For branching meshes, handle fills and strokes separately
+						let fill_regions = vector.find_closed_regions();
+						let has_fill = vector.style.fill() != &Fill::None;
+						let has_stroke = vector.style.stroke().is_some_and(|stroke| stroke.has_renderable_stroke());
+
+						// Render fills for each region separately
+						if has_fill {
+							for region in &fill_regions {
+								let mut bezpath = region.to_bezpath(vector);
+								bezpath.apply_affine(Affine::new(applied_stroke_transform.to_cols_array()));
+
+								do_fill_path(scene, &bezpath);
+							}
+						}
 
-							do_stroke(scene, 2.);
+						// Render stroke once for the entire mesh
+						if has_stroke {
+							let mut combined_path = kurbo::BezPath::new();
+							for mut bezpath in vector.stroke_bezpath_iter() {
+								bezpath.apply_affine(Affine::new(applied_stroke_transform.to_cols_array()));
+								for element in bezpath {
+									combined_path.push(element);
+								}
+							}
 
-							scene.pop_layer();
-							scene.pop_layer();
+							do_stroke_path(scene, &combined_path, 1.);
+						}
 
-							do_fill(scene);
-						} else {
-							// Fill first (unclipped), then stroke (clipped) above
-							do_fill(scene);
+						// If no fill and no stroke, still render something
+						if !has_fill && !has_stroke {
+							let mut combined_path = kurbo::BezPath::new();
+							for region in &fill_regions {
+								let mut bezpath = region.to_bezpath(vector);
+								bezpath.apply_affine(Affine::new(applied_stroke_transform.to_cols_array()));
+								for element in bezpath {
+									combined_path.push(element);
+								}
+							}
 
-							scene.push_layer(peniko::Mix::Normal, 1., kurbo::Affine::IDENTITY, &rect);
-							vector_table.render_to_vello(scene, parent_transform, _context, &render_params.for_alignment(applied_stroke_transform));
-							scene.push_layer(peniko::BlendMode::new(peniko::Mix::Normal, compose), 1., kurbo::Affine::IDENTITY, &rect);
+							// Use the existing rendering logic for this fallback case
```

Comment:
What do you mean by the "existing rendering logic"? In the diff, it appears to be completely new code.